### ISSUE TITLE
Crash while importing azuredevops_serviceendpoint_github.github_serviceendpoint: "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by CraigStuntzWillowTree** [microsoft/terraform-provider-azuredevops#250]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

```$ terraform -v
Terraform v0.13.4
+ provider registry.terraform.io/hashicorp/azuread v1.0.0
+ provider registry.terraform.io/hashicorp/azurerm v2.33.0
+ provider registry.terraform.io/microsoft/azuredevops v0.1.0
```

### Affected Resource(s)

* `azuredevops_serviceendpoint_github`

### Terraform Configuration Files

```hcl
resource "azuredevops_project" "project" {
  name               = "Project Name"
  visibility         = "private"
  version_control    = "Git"
  work_item_template = "Agile"
}

resource "azuredevops_serviceendpoint_github" "github_serviceendpoint" {
  project_id            = azuredevops_project.project.id
  service_endpoint_name = "GitHub Service Connection"
}
```

Then I did:
```bash
$ terraform import  azuredevops_serviceendpoint_github.github_serviceendpoint "Project Name"/9d58df75-a030-4a2e-bc44-xxxxxxxxxxx
```

### Debug Output

https://gist.github.com/CraigStuntzWillowTree/6b75b48e95ec70a3a451300d548ad7c6

### Expected Behavior

Terraform should not crash

### Actual Behavior

Terraform crashed

### Steps to Reproduce

1. `$ terraform import  azuredevops_serviceendpoint_github.github_serviceendpoint "Project Name"/9d58df75-a030-4a2e-bc44-xxxxxxxxxxx` on an existing ADO project/GitHub repo

**Comment by xuzhang3** [comment]
Hi @CraigStuntzWillowTree  I cannot reproduce your error. Which type `github` service connection you try to import. Currently we only support `GitHub`

**Comment by CraigStuntzWillowTree** [comment]
I'm actually using GitHub, although I don't remember if I specified `Git` or `GitHub` in the HCL. Based on the above it looks like I specified `Git`. Regardless it feels like terraform should return an error stating that only GitHub is supported instead of crashing.

**Comment by xuzhang3** [comment]
Hi @CraigStuntzWillowTree  I cannot reproduce your error.  The  root error is `Authorization.Scheme` is empty, but  `Authorization.Scheme`  is a required parameter for service connections.  I will keep this issue opening in case you have more information or questions.

**Comment by CraigStuntzWillowTree** [comment]
Did you notice I had no `auth_[...]` parameter on the `azuredevops_serviceendpoint_github` at all? I wonder if that's connected? Again, I don't presume I'm doing this right, I might very well have omitted something important! I just want a better message about it.

**Comment by xuzhang3** [comment]
1.  `azuredevops_serviceendpoint_github`  cannot created without `auth[]`
2. All operation steps. How does the project and service connection created?  From Terraform or web page?  Have you modify the .tfstate file manually?


There is no errors in the following scenario:
1. Create project and service connection from web and using Terraform import them. 
2. Create project and service connection with Terraform, remove the resources from .tfstate and import.

**Comment by CraigStuntzWillowTree** [comment]
Again, perhaps the problem was that I had `auth_[...]` missing from `azuredevops_serviceendpoint_github`.

We manually created the ADO project and service connection, from the web page. I did not modify the `.tfstate` manually. I tried doing `terraform import` using _exactly the HCL above._ Perhaps I should have included an `auth_` section. I would expect an error message indicating this instead of a terraform crash.
